src/test/java/io/cdap/plugin/gcp/bigquery/connector/BigQueryConnectorTest.java

```diff
@@ -35,85 +45,101 @@
 import io.cdap.plugin.gcp.bigquery.sink.BigQuerySink;
 import io.cdap.plugin.gcp.bigquery.source.BigQuerySource;
 import io.cdap.plugin.gcp.bigquery.sqlengine.BigQuerySQLEngine;
+import io.cdap.plugin.gcp.common.GCPUtils;
+import io.cdap.plugin.gcp.common.TestEnvironment;
+import org.junit.AfterClass;
 import org.junit.Assert;
-import org.junit.Assume;
 import org.junit.BeforeClass;
 import org.junit.Test;
 
-import java.io.File;
 import java.io.IOException;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Paths;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 /**
- * BigQuery Connector integration test. This test will only be run when below property is provided:
- * project.id -- the name of the project where temporary table or staging bucket may be created or BigQuery job may
- * be run. It will default to active google project if you have google cloud client installed.
- * dataset.project -- optional, the name of the project where the dataset is
- * dataset.name -- the name of the dataset
- * table.name -- the name of the table
- * service.account.file -- the path to the service account key file
+ * BigQuery Connector integration test.
+ *
+ * The service account used to run this test needs BigQuery admin permissions in the project.
  */
 public class BigQueryConnectorTest {
   private static final Set<String> SUPPORTED_TYPES = new HashSet<>(Arrays.asList("table", "view"));
-  private static String serviceAccountKey;
+  private static TestEnvironment testEnvironment;
   private static String project;
   private static String datasetProject;
   private static String dataset;
   private static String table;
-  private static String serviceAccountFilePath;
+  private static BigQuery bigQuery;
+  private static final Schema SCHEMA =
+    Schema.recordOf("output",
+                    Schema.Field.of("id", Schema.nullableOf(Schema.of(Schema.Type.LONG))),
+                    Schema.Field.of("name", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+  private static final Set<StructuredRecord> TABLE_DATA = new HashSet<>(Arrays.asList(
+    StructuredRecord.builder(SCHEMA).set("id", 0L).set("name", "alice").build(),
+    StructuredRecord.builder(SCHEMA).set("id", 1L).set("name", "bob").build()));
 
   @BeforeClass
   public static void setupTestClass() throws Exception {
-    // Certain properties need to be configured otherwise the whole tests will be skipped.
-
-    String messageTemplate = "%s is not configured, please refer to javadoc of this class for details.";
+    testEnvironment = TestEnvironment.load();
+
+    project = testEnvironment.getProject();
+    datasetProject = testEnvironment.getProject();
+    dataset = "bq_connector_test_" + System.currentTimeMillis();
```

Comment:
can we use UUID here to avoid potential race condition where two builds trigger at the same time?

Reply:
we can if we strip out the dashes (not an allowed character), though it becomes a bit more difficult visually to tell if it is an old dataset that is safe to delete or if it is currently in use. Let me see if I can use a uuid plus a timestamp.

Reply:
changed to use timestamp in seconds followed by uuid. So it's still sorted, but also unique.